### ISSUE TITLE
Scratchy Audio after boot unless I manually restart pipewire/wireplumber

### ISSUE THREAD

**Issue by zono50** [wwmm/easyeffects#3489]
### EasyEffects Version

7.1.8

### What package are you using?

Arch (easyeffects)

### Distribution

Arch Linux

### Describe the bug

getting the following error when trying to autostart easyeffects - arting easyeffects - app-com.github.wwmm.easyeffects@autostart.service: failed with result 'exit-code'
app-com.github.wwmm.easyeffects@autostart.service: main process exited, code=exited, status=1/FAILURE
failed with result 'core-dump'
main process exited, code=dumped, status=6/ABRT
pipe_manager.cpp remote error message: connection error
pipe_manager.cpp: 1513 Remote error res: Broken pipe
pipe_manager:cpp:1514 Remote error message: unknown resource 34 op:2
and then get clipping is broken, everything is clipped, but we didn't early-exit

after that, audio is scratchy and distorted until i restart pipewire and wireplumber

i've tried setting it up as a systemd service under user, i've tried reinstalling, and not sure what the issue is. Here is my list of plugins i'm using as well.

![image](https://github.com/user-attachments/assets/3be2b387-18fa-41c0-b32b-d3d9d62cef53)

![image](https://github.com/user-attachments/assets/2b758375-2c79-4492-90e2-a0765e24ba77)

### Expected Behavior

I expect easyeffects to load and my audio to work properly.

### Debug Log

These are the following logs i've accumulated over my testing - 

starting easyeffects - app-com.github.wwmm.easyeffects@autostart.service: failed with result 'exit-code'
app-com.github.wwmm.easyeffects@autostart.service: main process exited, code=exited, status=1/FAILURE
failed with result 'core-dump'
main process exited, code=dumped, status=6/ABRT
pipe_manager.cpp remote error message: connection error
pipe_manager.cpp: 1513 Remote error res: Broken pipe
pipe_manager:cpp:1514 Remote error message: unknown resource 34 op:2
and then get clipping is broken, everything is clipped, but we didn't early-exit

Nov 08 12:27:21 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) xrun state:0x7fc9160ec008 pending:2/19 s:174251999333 a:17>
Nov 08 12:27:21 CachyBox pipewire[1018]: pw.node: (ee_soe_output_level-87) xrun state:0x7fc915370008 pending:0/2 s:174254731246 a:174254733454 f:17425324462>
Nov 08 12:27:23 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) graph xrun not-triggered (191 suppressed)
Nov 08 12:27:23 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) xrun state:0x7fc9160ec008 pending:4/19 s:176252733352 a:17>
Nov 08 12:27:23 CachyBox pipewire[1018]: pw.node: (ee_soe_autogain-183) xrun state:0x7fc91490d008 pending:0/2 s:176255058679 a:176255060503 f:176253568076 w>
Nov 08 12:27:27 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) graph xrun not-triggered (5 suppressed)
Nov 08 12:27:27 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) xrun state:0x7fc9160ec008 pending:0/19 s:180410606556 a:18>
Nov 08 12:27:40 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) graph xrun not-triggered (0 suppressed)
Nov 08 12:27:40 CachyBox pipewire[1018]: pw.node: (alsa_output.pci-0000_03_00.0.iec958-stereo-56) xrun state:0x7fc9160ec008 pending:2/19 s:193475497288 a:19>
Nov 08 12:27:40 CachyBox pipewire[1018]: pw.node: (ee_soe_output_level-87) xrun state:0x7fc915370008 pending:0/2 s:193478207952 a:193478209579 f:19347664641

### Additional Information

i'm getting all kinds of errors on boot when i boot up easy effects, I had it set as an autostart application, and it would load, but i would get all kinds of errors. When i go into autostart, it also says failed. I decided to create a systemd service for it, and it's still failing. only thing that stops the audio errors is to restart pipewire pipewire pulse and wireplumber.

**Comment by violetmage** [comment]
It looks like you're on cachyOS, is that right?

If so, please ensure all your packages are up to date, as there was a somewhat recently released update to ananicy rules in cachyos that fixed an interference with easyeffects.

Also, just to confirm, you're _not_ using a flatpak version of easyeffects?

**Comment by zono50** [comment]
Yes I am on cachyos.

I update packages daily, but i will push another update for now to confirm.

correct i installed easyeffects through pacman

**Comment by violetmage** [comment]
If easyeffects is completely closed, and then restarted, everything works normally?

Is the easyeffects option 'Shutdown on Window Closing' turned on?

**Comment by zono50** [comment]
Nope, still doesn't work. I have to restart pipewire and wireplumber services under the system user folder before it actss right. Now i did create my own pipewire.service and wireplumber.service files under the user folder, put a 5 second delay, and rebooted, and that seemed to work. i'm going to reboot and test again to make sure it wasn't a fluke.

**Comment by violetmage** [comment]
Maybe your window manager is trying to be in control of hardware volume?

Does your audio work normally if easyeffects isn't autostarted at all?

**Comment by zono50** [comment]
i'm not too worried about it. this resolved the issue for me. Create a pipewire.service and wireplumber.service in the ~/.config/systemd/user folder.

  GNU nano 8.2                                                          pipewire.service                                                                     
[Unit]
Description=PipeWire Multimedia System
After=network.target default.target

[Service]
ExecStart=/usr/bin/pipewire
Restart=always
RestartSec=5

[Install]
WantedBy=default.target


for wireplumber.service - 

  GNU nano 8.2                                                         wireplumber.service                                                                   
[Unit]
Description=WirePlumber PipeWire Session Manager
After=pipewire.service default.target

[Service]
ExecStart=/usr/bin/wireplumber
Restart=always
RestartSec=5

[Install]
WantedBy=default.target

Maybe the system was booting too fast and the services couldn't catch up, i'm not sure. i'm kinda scared to delete the files since it's working. I will test some more tomorrow as i get off work soon, and my son likes to play on my pc.

**Comment by wwmm** [comment]
>Maybe the system was booting too fast and the services couldn't catch up, i'm not sure.

If EasyEffects starts before PiepeWire it definitely won't work. But killing easyeffects `easyeffects -q` and restarting it should be enough to fix the problem. Having to restart PipeWire is strange. Maybe this is related to the very low quantum (latency) that is in the pw-top output. If the hardware is not able to properly handle this the "scratchy audio" is explained.

**Comment by zono50** [comment]
i'm not too worried about it. this resolved the issue for me. Create a pipewire.service and wireplumber.service in the ~/.config/systemd/user folder.

  GNU nano 8.2                                                          pipewire.service                                                                     
[Unit]
Description=PipeWire Multimedia System
After=network.target default.target

[Service]
ExecStart=/usr/bin/pipewire
Restart=always
RestartSec=5

[Install]
WantedBy=default.target


for wireplumber.service - 

  GNU nano 8.2                                                         wireplumber.service                                                                   
[Unit]
Description=WirePlumber PipeWire Session Manager
After=pipewire.service default.target

[Service]
ExecStart=/usr/bin/wireplumber
Restart=always
RestartSec=5

[Install]
WantedBy=default.target

Maybe the system was booting too fast and the services couldn't catch up, i'm not sure. i'm kinda scared to delete the files since it's working. I will test some more tomorrow as i get off work soon, and my son likes to play on my pc.

**Comment by zono50** [comment]
Interestingly enough this morning, the problem surfaced itself again

**Comment by zono50** [comment]
Interestingly enough this morning, the problem surfaced itself again. I rebooted it this morning and now it's fine. I can't sseem to replicate it, rebooted it 4 times lol.

**Comment by wwmm** [comment]
> Interestingly enough this morning, the problem surfaced itself again. I rebooted it this morning and now it's fine. I can't sseem to replicate it, rebooted it 4 times lol.

Maybe a solution would be to somehow delay EasyEffects autostart. I do not remember if desktop files allow this or if a custom script is needed.

**Comment by wwmm** [comment]
So much has changed that is probably better to close this issue
